### PR TITLE
Make fee market instantiable

### PR DESCRIPTION
**Pull request by AsceticBear**
Companion for https://github.com/darwinia-network/darwinia-common/pull/1171

https://github.com/darwinia-network/darwinia-common/pull/1158 This might be helpful to review.

### CODE REVIEW

File: modules/fee-market/rpc/runtime-api/src/lib.rs [hunk 1]
```diff
@@ -73,9 +73,7 @@ decl_runtime_apis! {
 	where
 		Balance: Codec + MaybeDisplay + MaybeFromStr,
 	 {
-		fn market_fee(
-		) -> Option<Fee<Balance>>;
-
-		fn in_process_orders() -> InProcessOrders;
+		fn market_fee(instance: u8) -> Option<Fee<Balance>>;
```
Comment (wuminzhe) [inline modules/fee-market/rpc/runtime-api/src/lib.rs:76]: market fee api will be removed. keep it here?

Comment (AsceticBear) [reply modules/fee-market/rpc/runtime-api/src/lib.rs:76]: Keep it here right now. I'll clean it after the storage-read functionality is stable on the SDK side.

File: modules/fee-market/src/s2s/payment.rs [hunk 2]
```diff
@@ -41,19 +41,19 @@ pub struct FeeMarketPayment<T, I, Currency> {
 	_phantom: sp_std::marker::PhantomData<(T, I, Currency)>,
 }
 
-impl<T, I, Currency> MessageDeliveryAndDispatchPayment<T::Origin, T::AccountId, RingBalance<T>>
+impl<T, I, Currency> MessageDeliveryAndDispatchPayment<T::Origin, T::AccountId, RingBalance<T, I>>
 	for FeeMarketPayment<T, I, Currency>
```
Comment (wuminzhe) [inline modules/fee-market/src/s2s/payment.rs:45]: There is a `RootAccount` in the darwina-common. `FeeMarketPayment<T, I, Currency, RootAccount>`.
what is the difference?

Comment (AsceticBear) [reply modules/fee-market/src/s2s/payment.rs:45]: Good catch. The main branch of `darwinia-bridges-substrate` repo includes such a change https://github.com/paritytech/parity-bridges-common/pull/962, and `darwinia-common` not yet. That makes the difference.

Comment (wuminzhe) [reply modules/fee-market/src/s2s/payment.rs:45]: thanks

**Review by wuminzhe — APPROVED** [review]

**Comment by HackFisher** [comment]
There are conflicts

File: modules/fee-market/rpc/runtime-api/src/lib.rs (hunk 1 above)
Comment (HackFisher) [reply modules/fee-market/rpc/runtime-api/src/lib.rs:76]: After confirm with apps and bridger, these rpc can be safely removed.

**Comment by AsceticBear** [comment]
@HackFisher Please take a review again.

**Review by HackFisher — APPROVED** [review]

**Pull request merged**